### PR TITLE
Add NDAX REST Api Support

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

**Comment by vslee** [comment]
Since this PR uses different JSON parsing logic than the rest of the library, I'll let jjxrta review it.

**Comment by vslee** [comment]
Can simplify the namespaces a bit though. For example, the converters and models can be in ExchangeSharp.NDAX. Also, looks like the official capitalization is NDAX.

File: ExchangeSharp/API/Exchanges/Ndax/Converters/BoolConverter.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+using System;
+using Newtonsoft.Json;
+
+namespace ExchangeSharp.API.Exchanges.Ndax.Converters
```
Comment (vslee) [inline ExchangeSharp/API/Exchanges/Ndax/Converters/BoolConverter.cs]: simplify to ExchangeSharp.NDAX

File: ExchangeSharp/API/Exchanges/Ndax/Models/GenericResponse.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+using Newtonsoft.Json;
+
+namespace ExchangeSharp.API.Exchanges.Ndax.Models
```
Comment (vslee) [inline ExchangeSharp/API/Exchanges/Ndax/Models/GenericResponse.cs]: simplify to ExchangeSharp.NDAX

File: ExchangeSharp/API/Exchanges/Ndax/ExchangeNdaxAPI.cs [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+
+            return null;
+        }
+
+    }
+
+
+    public partial class ExchangeName
+    {
+        public const string Ndax = "Ndax";
```
Comment (vslee) [inline ExchangeSharp/API/Exchanges/Ndax/ExchangeNdaxAPI.cs]: From their site, the capitalization is NDAX. Conflicts slightly w/ C# capitalization guidelines, but I've been changing the exchange capitalization to match their official titles.

**Comment by jjxtra** [comment]
Let's make all those model objects use internal instead of public, I don't think anyone outside of the internals of this exchange will care to see them. Even better, make the exchange a partial class and embed the models inside of the class.

**Comment by jjxtra** [comment]
Binance uses similar json parsing and model object strategy so it is fine. Depending on complexity of Exchange model objects can be helpful.

File: ExchangeSharp/API/Exchanges/NDAX/Converters/BoolConverter.cs [hunk 4]
```diff
@@ -1,7 +1,7 @@
 using System;
 using Newtonsoft.Json;
 
-namespace ExchangeSharp.API.Exchanges.Ndax.Converters
+namespace ExchangeSharp.NDAX
 {
     public class BoolConverter : JsonConverter
     {
```
Comment (jjxtra) [inline ExchangeSharp/API/Exchanges/NDAX/Converters/BoolConverter.cs]: This can be an internal class, like the others

**Comment by Kukks** [comment]
Thanks guys, I'll do a followup PR with partial websocket support and any cleanup you want( just leave a comment here) tonight. :)

**Pull request merged**